### PR TITLE
AppSSO: Service Operator can configure AuthServer storage

### PR DESCRIPTION
**Pull request by ddubson**
Which other branches should this be merged with (if any)?

### CODE REVIEW

File: app-sso/crds/authserver.md [hunk 1]
```diff
@@ -183,7 +192,10 @@ to troubleshoot issues.
 `.status.deployments.authServer` describes the current authorization server deployment by listing the running image,
 its replicas, the hash of the current configuration and the generation which has last resulted in a restart.
 
-`.status.deployments.redis` describes the current Redis deployment by listing its running image.
+`.status.deployments.redis` describes the current provided Redis deployment by listing its running image. This field is
+nil if storage is defined explicitly via 'spec.storage'.
```
Comment (mamachanko) [inline app-sso/crds/authserver.md:196]:
```suggestion
nil if storage is defined explicitly via '.spec.storage'.
```
🤓

File: app-sso/service-operators/configuring-storage.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Configuring storage for an `AuthServer`
```
Comment (mamachanko) [inline app-sso/service-operators/configuring-storage.md:1]: **[suggestion]:** To be consistent with existing section name, I suggest
```suggestion
# Storage
```

File: app-sso/service-operators/configuring-storage.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+AppSSOs `AuthServer` handles data pertaining to user's session, security tokens and codes, and approved or rejected
```
Comment (mamachanko) [inline app-sso/service-operators/configuring-storage.md:3]: **[thought]:** "security tokens and codes" may be vague or imprecise. We could replace with the OAuth2-specific terms.

File: app-sso/service-operators/configuring-storage.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+consents. For production environments, it is critical to provide your own storage source to enable critical enterprise
```
Comment (mamachanko) [inline app-sso/service-operators/configuring-storage.md:4]: **[suggestion, style]:** Say "critical" only once.

File: app-sso/service-operators/configuring-storage.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+## Storage provided by default
+
+---
+💡Storage provided by default refers to an `AuthServer` resource in which the field `.spec.storage` is not set.
+---
+
+---
+⚠️ The default storage configuration is most useful in prototyping or testing environments, and **should not be relied
+on in production environments.**.
+---
```
Comment (mamachanko) [inline app-sso/service-operators/configuring-storage.md:17]: **[question]:** Wouldn't quote blocks be the Markdown idiom here?

Comment (ddubson) [reply app-sso/service-operators/configuring-storage.md]: Changed this to the proper _Notes_ feature (the real blockquote)

File: app-sso/getting-started/provision-auth-server.md [hunk 6]
```diff
@@ -226,3 +226,11 @@ consider using OpenID Connect IDPs instead.
 
 The `email` and `roles` fields are optional for internal users. However, they will be useful when we want to use SSO
 with a client application later in this guide.
+
+### Configuring storage
+
+An `AuthServer` has in-memory storage built-in, and may be used for testing, prototyping, and other non-production
```
Comment (mamachanko) [inline app-sso/getting-started/provision-auth-server.md:232]: **[thought]:** I find "`AuthServer` has in-memory storage built-in" misleading.

File: app-sso/service-operators/storage.hbs.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
+# Storage
+
+AppSSOs `AuthServer` handles data pertaining to user's session, identity and access tokens, and approved or rejected
+consents. For production environments, it is critical to provide your own storage source to enable enterprise
```
Comment (mamachanko) [inline app-sso/service-operators/storage.hbs.md:4]: **[praise]:** 🎯 

> user's session, identity and access tokens, and approved or rejected consents

Comment (ddubson) [reply app-sso/service-operators/storage.hbs.md:4]: ❤️

File: app-sso/service-operators/storage.hbs.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  username: "redis01-user"           # optional
+```
+
+### Attaching storage to an AuthServer
+
+Once a Redis Secret resource is applied, you may reference the Secret in `.spec.storage`. Here is an example of an
+AuthServer with a reference to a Redis Secret:
+
+```yaml
+kind: AuthServer
```
Comment (mamachanko) [inline app-sso/service-operators/storage.hbs.md:92]:
```suggestion
apiVersion: sso.apps.tanzu.vmware.com/v1alpha1
kind: AuthServer
```

Comment (ddubson) [reply app-sso/service-operators/storage.hbs.md:92]: Added

File: app-sso/crds/authserver.md [hunk 9]
```diff
@@ -1,7 +1,7 @@
 # AuthServer
 
 `AuthServer` represents the request for an OIDC authorization server. It results in the deployment of an authorization
-server backed by Redis over mTLS.
+server backed by Redis over mutual TLS (if no external storage is explicitly configured).
```
Comment (candicenonsense) [inline app-sso/crds/authserver.md:5]: Let's use language more like in our Arch Review and remove 'external' here and below - "if no storage is defined/configured" is fine and shorter.

File: app-sso/crds/authserver.md [hunk 10]
```diff
@@ -212,8 +224,12 @@ deployments:
     configHash: "11216479096262796218"
     image: "..."
     replicas: 1
-  redis:
+  redis: # empty if external storage is defined
     image: "..."
```
Comment (candicenonsense) [inline app-sso/crds/authserver.md:228]: "if storage is configured by the operator"

Comment (ddubson) [reply app-sso/crds/authserver.md:228]: Added

File: app-sso/service-operators/storage.hbs.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+```bash
+kubectl get authserver <authserver-name> \
+  --namespace <authserver-namespace> \
+  --output jsonpath="{.status.storage.redis}" | jq
+```
+
+## Storage provided by default
+
+If no external storage is defined, an `AuthServer` provides its own short-lived ephemeral storage solution,
```
[the remark this answers is not in the document]
Comment (ddubson) [reply app-sso/service-operators/storage.hbs.md]: Added

File: app-sso/crds/authserver.md (hunk 9 above)
Comment (ddubson) [reply app-sso/crds/authserver.md:5]: Changed to simpler language.

File: app-sso/getting-started/provision-auth-server.md (hunk 6 above)
Comment (ddubson) [reply app-sso/getting-started/provision-auth-server.md]: Changed.

**Review by candicenonsense — APPROVED** [review]

**Pull request closed without merging**